autofit/optimize/non_linear/nested_sampling/dynesty.py

```diff
@@ -0,0 +1,527 @@
+
+        self.paths.backup()
+
+        print(dynesty_output.pdf.getMeans())
+
+        instance = dynesty_output.most_likely_instance
+        dynesty_output.output_results(during_analysis=False)
+        return Result(
+            instance=instance,
+            likelihood=dynesty_output.maximum_log_likelihood,
+            output=dynesty_output,
+            previous_model=model,
+            gaussian_tuples=dynesty_output.gaussian_priors_at_sigma(self.sigma),
+        )
+
+    def output_from_model(self, model, paths):
+        """Create this non-linear search's output class from the model and paths.
+
+        This function is required by the aggregator, so it knows which output class to generate an instance of."""
+        return DynestyOutput(model=model, paths=paths)
+
+    def sampler_fom_model_and_fitness(self, model, fitness_function):
+        return NotImplementedError()
+
+
+class DynestyStatic(AbstractDynesty):
+    def __init__(self, paths=None, sigma=3):
+        """
+        Class to setup and run a Dynesty non-linear search, using the static Dynesty nested sampler described at this
+        webpage:
+
+        https://dynesty.readthedocs.io/en/latest/api.html#module-dynesty.dynesty
+
+        See the docstring for *AbstractDynesty* for complete documentation of Dynesty samplers.
+
+        Attributes
+        ----------
+        sigma : float
+            The error-bound value that linked Gaussian prior withs are computed using. For example, if sigma=3.0,
+            parameters will use Gaussian Priors with widths coresponding to errors estimated at 3 sigma confidence.
+        iterations_per_update : int
+            The number of iterations performed between every Dynesty back-up (via dumping the Dynesty instance as a
+            pickle).
+        """
+
+        super().__init__(paths=paths, sigma=sigma)
+
+        self.n_live_points = self.config("n_live_points", int)
+
+        logger.debug("Creating DynestyStatic NLO")
+
+    @property
+    def name(self):
+        return "dynesty_static"
+
+    def copy_with_name_extension(self, extension, remove_phase_tag=False):
+        """Copy this instance of the dynesty non-linear search with all associated attributes.
+
+        This is used to set up the non-linear search on phase extensions."""
+        copy = super().copy_with_name_extension(
+            extension=extension, remove_phase_tag=remove_phase_tag
+        )
+        copy.n_live_points = self.n_live_points
+
+        return copy
+
+    def sampler_fom_model_and_fitness(self, model, fitness_function):
+        """Get the static Dynesty sampler which performs the non-linear search, passing it all associated input Dynesty
+        variables."""
+
+        return StaticSampler(
+            loglikelihood=fitness,
+            prior_transform=prior,
+            ndim=model.prior_count,
+            logl_args=[model, fitness_function],
+            ptform_args=[model],
+            nlive=self.n_live_points,
```

Comment:
Golly that's a lot of arguments

Reply:
Yep, these non-linear samplers are... complicated.